processing/src/test/java/io/druid/granularity/QueryGranularityTest.java

```diff
@@ -642,6 +643,13 @@ public void testSerializePeriod() throws Exception
     String jsonOut = mapper.writeValueAsString(expected);
     Assert.assertEquals(expected, mapper.readValue(jsonOut, QueryGranularity.class));
 
+    String illegalJson = "{ \"type\": \"period\", \"period\": \"P0D\" }";
+    try {
+      mapper.readValue(illegalJson, QueryGranularity.class);
```

Comment:
suggest `Assert.fail()` here just to make sure the exception is actually thrown. Or, use the `@Rule` ExpectedException (example in FileTaskLogsTest)